### PR TITLE
CM-20731 Cycode CLI - print update status to console by printer

### PR DESCRIPTION
**Pull request by morsa4406**

### CODE REVIEW

File: cli/code_scanner.py [hunk 1]
```diff
@@ -367,38 +390,49 @@ def perform_scan(cycode_client, zipped_documents: InMemoryZip, scan_type: str, s
...
+    last_scan_update_at = None
+    output_type = context.obj['output']
     end_polling_time = time.time() + polling_timeout
+    spinner = Halo(spinner='dots')
+    spinner.start("Scan in progress")
     while time.time() < end_polling_time:
         logger.debug("scan in progress")
         scan_details = cycode_client.get_scan_details(scan_id)
+        if scan_details.scan_update_at is not None and scan_details.scan_update_at != last_scan_update_at:
+            last_scan_update_at = scan_details.scan_update_at
```
Comment (MichalBor) [inline cli/code_scanner.py:425]: updated_at

File: cli/consts.py [hunk 2]
```diff
@@ -82,7 +82,7 @@
 SCA_ZIP_MAX_SIZE_LIMIT_IN_BYTES = 209715200
 
 # scan with polling
-SCAN_POLLING_WAIT_INTERVAL_IN_SECONDS = 10
+SCAN_POLLING_WAIT_INTERVAL_IN_SECONDS = 2
```
Comment (MichalBor) [inline cli/consts.py:85]: why?

File: cli/printers/base_printer.py [hunk 3]
```diff
@@ -12,5 +13,9 @@ def __init__(self, context: click.Context):
         self.context = context
 
     @abstractmethod
-    def print_results(self, results: List[DocumentDetections]):
+    def print_results(self, context: click.Context, results: List[DocumentDetections]):
```
Comment (MichalBor) [inline cli/printers/base_printer.py:16]: merge from main

File: cli/printers/text_printer.py [hunk 4]
```diff
@@ -38,6 +42,17 @@ def print_results(self, results: List[DocumentDetections]):
         if self.context.obj.get('report_url'):
             click.secho(f"Report URL: {self.context.obj.get('report_url')}")
 
+    def print_scan_status(self, scan_details_response: models.ScanDetailsResponse):
+        click.secho("")
+        update_time = parser().parse(scan_details_response.scan_update_at).time().isoformat(timespec="seconds")
```
Comment (MichalBor) [inline cli/printers/text_printer.py:47]: move to function

File: cli/printers/text_printer.py [hunk 5]
```diff
@@ -38,6 +42,17 @@ def print_results(self, results: List[DocumentDetections]):
         if self.context.obj.get('report_url'):
             click.secho(f"Report URL: {self.context.obj.get('report_url')}")
 
+    def print_scan_status(self, scan_details_response: models.ScanDetailsResponse):
+        click.secho("")
```
Comment (MichalBor) [inline cli/printers/text_printer.py:46]: `click.echo()`

File: cli/printers/text_printer.py [hunk 6]
```diff
@@ -38,6 +42,17 @@ def print_results(self, results: List[DocumentDetections]):
         if self.context.obj.get('report_url'):
             click.secho(f"Report URL: {self.context.obj.get('report_url')}")
 
+    def print_scan_status(self, scan_details_response: models.ScanDetailsResponse):
```
Comment (MichalBor) [inline cli/printers/text_printer.py:45]: import ScanDetailsResponse

Comment (MichalBor) [inline cli/printers/text_printer.py:45]: rename to scan_details

File: cli/printers/text_printer.py [hunk 7]
```diff
@@ -38,6 +42,17 @@ def print_results(self, results: List[DocumentDetections]):
...
+        click.secho("")
+        update_time = parser().parse(scan_details_response.scan_update_at).time().isoformat(timespec="seconds")
+        self._print_click_secho_with_time(update_time, f"Scan update: (scan_id: {scan_details_response.id})")
+        self._print_click_secho_with_time(update_time, f"Scan status: {scan_details_response.scan_status}")
+        if scan_details_response.message is not None:
+            self._print_click_secho_with_time(update_time, f"Scan message: {scan_details_response.message}")
+
+    def _print_click_secho_with_time(self, update_time, message):
+        click.secho(f"[{update_time}] {message}")
```
Comment (MichalBor) [inline cli/printers/text_printer.py:54]: its the same func you added to scan_utils?

File: cli/utils/scan_utils.py [hunk 8]
```diff
@@ -1,7 +1,14 @@
...
 
 
 def is_scan_failed(context: click.Context):
     did_fail = context.obj.get("did_fail")
     issue_detected = context.obj.get("issue_detected")
     return did_fail or issue_detected
+
+
+def print_click_secho(message: str):
+    time = datetime.now().time().isoformat(timespec="seconds")
```
Comment (MichalBor) [inline cli/utils/scan_utils.py:13]: what you get here?

File: cli/utils/scan_utils.py [hunk 9]
```diff
@@ -1,7 +1,14 @@
...
 import click
 
 
 def is_scan_failed(context: click.Context):
     did_fail = context.obj.get("did_fail")
     issue_detected = context.obj.get("issue_detected")
     return did_fail or issue_detected
+
+
+def print_click_secho(message: str):
```
Comment (MichalBor) [inline cli/utils/scan_utils.py:12]: rename func

File: cli/code_scanner.py [hunk 10]
```diff
@@ -367,38 +390,49 @@ def perform_scan(cycode_client, zipped_documents: InMemoryZip, scan_type: str, s
     return scan_result
 
 
-def perform_scan_async(cycode_client, zipped_documents: InMemoryZip, scan_type: str,
+def perform_scan_async(context: click.Context, cycode_client, zipped_documents: InMemoryZip, scan_type: str,
```
Comment (MichalBor) [inline cli/code_scanner.py:393]: from where the context comes? you don't pass it, you pass the output_type

File: cli/code_scanner.py [hunk 11]
```diff
@@ -367,38 +390,49 @@ def perform_scan(cycode_client, zipped_documents: InMemoryZip, scan_type: str, s
...
-    return poll_scan_results(cycode_client, scan_async_result.scan_id, timeout)
+    return poll_scan_results(context, cycode_client, scan_async_result.scan_id, timeout)
 
 
-def poll_scan_results(cycode_client, scan_id: str, polling_timeout: int = None):
+def poll_scan_results(context: click.Context, cycode_client, scan_id: str, polling_timeout: int = None):
     if polling_timeout is None:
         polling_timeout = configuration_manager.get_scan_polling_timeout_in_seconds()
 
+    printer = ResultsPrinter()
```
Comment (MichalBor) [inline cli/code_scanner.py:415]: ResultsPrinter().get_printer

File: cli/code_scanner.py [hunk 12]
```diff
@@ -610,10 +645,12 @@ def get_commit_range_modified_documents(path: str, from_commit_rev: str, to_comm
 
         file_content = sca_code_scanner.get_file_content_from_commit(repo, from_commit_rev, diff_file_path)
         if file_content is not None:
+            print_click_secho(f"Read file content from commit: {diff_file_path}")
```
Comment (MichalBor) [inline cli/code_scanner.py:648]: you can just use the logger instead

File: cli/utils/scan_utils.py (hunk 9 above)
Comment (MichalBor) [inline cli/utils/scan_utils.py:12]: not needed, use logger

**Review by MichalBor — CHANGES_REQUESTED** [review]

File: cli/utils/scan_utils.py (hunk 9 above)
Comment (morsa4406) [reply cli/utils/scan_utils.py:12]: It's should be printed always not by logger

File: cli/code_scanner.py (hunk 12 above)
Comment (morsa4406) [reply cli/code_scanner.py:648]: It's should be printed always not by logger by click.secho

File: cli/code_scanner.py (hunk 1 above)
Comment (morsa4406) [reply cli/code_scanner.py:424]: It's how it's called from DB:
image

File: cli/utils/scan_utils.py (hunk 9 above)
Comment (MichalBor) [reply cli/utils/scan_utils.py:12]: so use logger.info

File: cli/consts.py (hunk 2 above)
Comment (morsa4406) [reply cli/consts.py:85]: Show status/updates from server

File: cli/utils/scan_utils.py (hunk 9 above)
Comment (morsa4406) [reply cli/utils/scan_utils.py:12]: No we show also progress bar it can't be by logger.info

File: cli/printers/text_printer.py (hunk 6 above)
Comment (morsa4406) [reply cli/printers/text_printer.py:45]: Fix

Comment (morsa4406) [reply cli/printers/text_printer.py]: Fix

File: cli/utils/scan_utils.py (hunk 9 above)
Comment (morsa4406) [reply cli/utils/scan_utils.py]: Fix

**Review by MichalBor — CHANGES_REQUESTED** [review]

File: cli/utils/scan_utils.py [hunk 13]
```diff
@@ -1,7 +1,15 @@
...
+from cyclient import logger
+
 
 def is_scan_failed(context: click.Context):
     did_fail = context.obj.get("did_fail")
     issue_detected = context.obj.get("issue_detected")
     return did_fail or issue_detected
+
+
+def print_audit(message: str):
```
[the remark this answers is not in the document]
Comment (morsa4406) [reply cli/utils/scan_utils.py:14]: Fixed

File: cli/printers/text_printer.py [hunk 14]
```diff
@@ -38,6 +43,15 @@ def print_results(self, results: List[DocumentDetections]):
...
             click.secho(f"Report URL: {self.context.obj.get('report_url')}")
 
+    def print_scan_details(self, scan_details_response: ScanDetailsResponse):
+        self._audit_info_log(f"Scan update: (scan_id: {scan_details_response.id})")
+        self._audit_info_log(f"Scan status: {scan_details_response.scan_status}")
+        if scan_details_response.message is not None:
+            self._audit_info_log(f"Scan message: {scan_details_response.message}")
+
+    def _audit_info_log(self, message):
+        logger.info(message)
```
[the remark this answers is not in the document]
Comment (morsa4406) [reply cli/printers/text_printer.py:53]: Fixed move code to code_scanner.py

File: cli/code_scanner.py [hunk 15]
```diff
@@ -876,11 +912,15 @@ def wait_for_detections_creation(cycode_client, scan_id: str, expected_detection
     logger.debug("waiting for detections to be created")
     scan_persisted_detections_count = 0
     end_polling_time = time.time() + DETECTIONS_COUNT_VERIFICATION_TIMEOUT_IN_SECONDS
+    spinner = Halo(spinner='dots')
+    spinner.start("Please wait until printing scan result...")
```
[the remark this answers is not in the document]
Comment (morsa4406) [reply cli/code_scanner.py:916]: It's approved with Rony

**Comment by morsa4406** [comment]
@MichalBor Please, reminder

File: cli/code_scanner.py [hunk 16]
```diff
@@ -244,8 +251,11 @@ def scan_documents(context: click.Context, documents_to_scan: List[Document], is
     zipped_documents = InMemoryZip()
 
     try:
+        logger.info("Preparing local files")
```
Comment (MichalBor) [inline cli/code_scanner.py:254]: debug

File: cli/code_scanner.py [hunk 17]
```diff
@@ -328,19 +340,26 @@ def handle_scan_result(context, scan_result, command_scan_type, scan_type, sever
 def perform_pre_scan_documents_actions(context: click.Context, scan_type: str, documents_to_scan: List[Document],
                                        is_git_diff: bool = False):
     if scan_type == SCA_SCAN_TYPE:
+        logger.info(
```
Comment (MichalBor) [inline cli/code_scanner.py:343]: debug

File: cli/code_scanner.py [hunk 18]
```diff
@@ -328,19 +340,26 @@ def handle_scan_result(context, scan_result, command_scan_type, scan_type, sever
...
     start_zip_creation_time = time.time()
+    logger.info("Zipping documents:")
     for index, document in enumerate(documents):
         zip_file_size = getsizeof(zip.in_memory_zip)
         validate_zip_file_size(scan_type, zip_file_size)
 
         logger.debug('adding file to zip, %s',
                      {'index': index, 'filename': document.path, 'unique_id': document.unique_id})
         zip.append(document.path, document.unique_id, document.content)
+        logger.info(document.path)
```
Comment (MichalBor) [inline cli/code_scanner.py:360]: remove, you already have this info in the log above

File: cli/code_scanner.py [hunk 19]
```diff
@@ -543,6 +578,7 @@ def exclude_irrelevant_documents_to_scan(context: click.Context, documents_to_sc
         List[Document]:
     scan_type = context.obj['scan_type']
     logger.debug("excluding irrelevant documents to scan")
+    logger.info("Excluding irrelevant documents to scan")
```
Comment (MichalBor) [inline cli/code_scanner.py:581]: remove

File: cli/code_scanner.py [hunk 20]
```diff
@@ -622,10 +658,12 @@ def get_commit_range_modified_documents(path: str, from_commit_rev: str, to_comm
 
         file_content = sca_code_scanner.get_file_content_from_commit(repo, from_commit_rev, diff_file_path)
         if file_content is not None:
+            logger.info(f"Read file content from commit: {diff_file_path}")
             from_commit_documents.append(Document(file_path, file_content))
 
         file_content = sca_code_scanner.get_file_content_from_commit(repo, to_commit_rev, diff_file_path)
         if file_content is not None:
+            logger.info(f"Read file content to commit: {diff_file_path}")
```
Comment (MichalBor) [inline cli/code_scanner.py:666]: remove

File: cli/printers/base_printer.py [hunk 21]
```diff
@@ -12,5 +14,5 @@ def __init__(self, context: click.Context):
...
```
Comment (MichalBor) [inline cli/printers/base_printer.py:17]: the code changed. please merge from dev!

File: cli/printers/text_printer.py [hunk 22]
```diff
@@ -1,10 +1,12 @@
-import click
 import math
 from typing import List, Optional
-from cli.printers.base_printer import BasePrinter
-from cli.models import DocumentDetections, Detection, Document
+
+import click
```
Comment (MichalBor) [inline cli/printers/text_printer.py:4]: revert the changes in this file

**Review by MichalBor — COMMENTED** [review]
🚀

File: cli/printers/json_printer.py [hunk 23]
```diff
@@ -1,8 +1,10 @@
 import json
-import click
 from typing import List
-from cli.printers.base_printer import BasePrinter
+
+import click
+
 from cli.models import DocumentDetections
+from cli.printers.base_printer import BasePrinter
```
[the remark this answers is not in the document]
Comment (morsa4406) [reply cli/printers/json_printer.py]: Fixed

File: cli/printers/text_printer.py (hunk 22 above)
Comment (morsa4406) [reply cli/printers/text_printer.py]: Fixed

File: cli/printers/base_printer.py (hunk 21 above)
Comment (morsa4406) [reply cli/printers/base_printer.py]: Fixed

File: cli/code_scanner.py (hunk 20 above)
Comment (morsa4406) [reply cli/code_scanner.py:666]: Why?

File: cli/code_scanner.py [hunk 24]
```diff
@@ -622,10 +658,12 @@ def get_commit_range_modified_documents(path: str, from_commit_rev: str, to_comm
 
         file_content = sca_code_scanner.get_file_content_from_commit(repo, from_commit_rev, diff_file_path)
         if file_content is not None:
+            logger.info(f"Read file content from commit: {diff_file_path}")
```
[the remark this answers is not in the document]
Comment (morsa4406) [reply cli/code_scanner.py:661]: Why?

File: cli/code_scanner.py [hunk 25]
```diff
@@ -328,19 +340,26 @@ def handle_scan_result(context, scan_result, command_scan_type, scan_type, sever
...
+        logger.info(
+            f"Perform pre scan document actions")
         sca_code_scanner.add_dependencies_tree_document(context, documents_to_scan, is_git_diff)
+        logger.info(
+            f"Perform pre scan document actions is done")
 
 
 def zip_documents_to_scan(scan_type: str, zip: InMemoryZip, documents: List[Document]):
     start_zip_creation_time = time.time()
+    logger.info("Zipping documents:")
```
[the remark this answers is not in the document]
Comment (morsa4406) [reply cli/code_scanner.py:352]: Why?

File: cli/code_scanner.py (hunk 17 above)
Comment (morsa4406) [reply cli/code_scanner.py:343]: I put it on info since I would like to update user on processing

File: cli/code_scanner.py (hunk 16 above)
Comment (morsa4406) [reply cli/code_scanner.py:254]: I put it on info since I would like to update user on processing

File: cli/code_scanner.py [hunk 26]
```diff
@@ -281,11 +291,13 @@ def scan_commit_range_documents(context: click.Context, from_documents_to_scan:
     try:
         scan_result = init_default_scan_result(str(scan_id))
         if should_scan_documents(from_documents_to_scan, to_documents_to_scan):
+            logger.info("Preparing from-commit zip")
             from_commit_zipped_documents = zip_documents_to_scan(scan_type, from_commit_zipped_documents,
                                                                  from_documents_to_scan)
+            logger.info("Preparing to-commit zip")
```
[the remark this answers is not in the document]
Comment (morsa4406) [reply cli/code_scanner.py:297]: I put it on info since I would like to update user on processing

**Comment by morsa4406** [comment]
ctp

File: cli/code_scanner.py (hunk 16 above)
Comment (morsa4406) [reply cli/code_scanner.py:254]: Fixed

File: cli/code_scanner.py (hunk 26 above)
[the remark this answers is not in the document]
Comment (morsa4406) [reply cli/code_scanner.py:297]: Fixed

File: cli/code_scanner.py (hunk 17 above)
Comment (morsa4406) [reply cli/code_scanner.py:343]: Fixed

File: cli/code_scanner.py [hunk 27]
```diff
@@ -328,19 +340,26 @@ def handle_scan_result(context, scan_result, command_scan_type, scan_type, sever
 def perform_pre_scan_documents_actions(context: click.Context, scan_type: str, documents_to_scan: List[Document],
                                        is_git_diff: bool = False):
     if scan_type == SCA_SCAN_TYPE:
+        logger.info(
+            f"Perform pre scan document actions")
         sca_code_scanner.add_dependencies_tree_document(context, documents_to_scan, is_git_diff)
+        logger.info(
```
[the remark this answers is not in the document]
Comment (morsa4406) [reply cli/code_scanner.py:346]: Fixed

File: cli/code_scanner.py (hunk 25 above)
[the remark this answers is not in the document]
Comment (morsa4406) [reply cli/code_scanner.py:352]: Fixed

File: cli/code_scanner.py (hunk 24 above)
[the remark this answers is not in the document]
Comment (morsa4406) [reply cli/code_scanner.py:661]: Fixed

File: cli/code_scanner.py (hunk 20 above)
Comment (morsa4406) [reply cli/code_scanner.py:666]: Fixed

File: cli/code_scanner.py (hunk 18 above)
Comment (MichalBor) [reply cli/code_scanner.py:358]: remove

File: cli/code_scanner.py [hunk 28]
```diff
@@ -192,6 +197,7 @@ def pre_receive_scan(context: click.Context, ignored_args: List[str]):
 
 
 def scan_sca_pre_commit(context: click.Context):
+    logger.info(f"Scan sca pre commit")
```
Comment (MichalBor) [inline cli/code_scanner.py:200]: remove

File: cli/code_scanner.py [hunk 29]
```diff
@@ -203,6 +209,7 @@ def scan_sca_pre_commit(context: click.Context):
 
 
 def scan_sca_commit_range(context: click.Context, path: str, commit_range: str):
+    logger.info(f"Scan sca commit range")
```
Comment (MichalBor) [inline cli/code_scanner.py:212]: remove

File: cli/code_scanner.py [hunk 30]
```diff
@@ -357,10 +374,12 @@ def validate_zip_file_size(scan_type, zip_file_size):
             raise ZipTooLargeError(ZIP_MAX_SIZE_LIMIT_IN_BYTES)
 
 
-def perform_scan(cycode_client, zipped_documents: InMemoryZip, scan_type: str, scan_id: UUID, is_git_diff: bool,
+def perform_scan(context, cycode_client, zipped_documents: InMemoryZip, scan_type: str, scan_id: UUID,
+                 is_git_diff: bool,
                  is_commit_range: bool, scan_parameters: dict):
+    logger.info("Perform scan")
```
Comment (MichalBor) [inline cli/code_scanner.py:380]: remove

File: cli/code_scanner.py (hunk 19 above)
Comment (MichalBor) [reply cli/code_scanner.py:579]: remove

File: cli/helpers/sca_code_scanner.py [hunk 31]
```diff
@@ -74,17 +76,20 @@ def add_dependencies_tree_document(context: click.Context, documents_to_scan: Li
...
-            gradle_dependencies_tree = try_generate_dependencies_tree(
-                get_manifest_file_path(document, is_monitor_action, project_path))
+            gradle_manifest_file_path = get_manifest_file_path(document, is_monitor_action, project_path)
+            gradle_dependencies_tree = try_generate_dependencies_tree(gradle_manifest_file_path)
             if gradle_dependencies_tree is None:
                 logger.warning('Error occurred while trying to generate gradle dependencies tree. %s',
                                {'filename': document.path})
                 documents_to_add.append(
                     Document(build_dep_tree_path(document.path, BUILD_GRADLE_DEP_TREE_FILE_NAME), '', is_git_diff))
+                logger.info(
```
Comment (MichalBor) [inline cli/helpers/sca_code_scanner.py:86]: remove or debug

File: cli/helpers/sca_code_scanner.py [hunk 32]
```diff
@@ -74,17 +76,20 @@ def add_dependencies_tree_document(context: click.Context, documents_to_scan: Li
...
+                    f"Failed to generate Gradle dependencies tree on path: {gradle_manifest_file_path}")
             else:
                 documents_to_add.append(
                     Document(build_dep_tree_path(document.path, BUILD_GRADLE_DEP_TREE_FILE_NAME),
                              gradle_dependencies_tree, is_git_diff))
+                logger.info(f"Succeeded to generate Gradle dependencies tree on path: {gradle_manifest_file_path}")
```
Comment (MichalBor) [inline cli/helpers/sca_code_scanner.py:92]: same

Comment (morsa4406) [reply cli/helpers/sca_code_scanner.py]: Fixed

File: cli/helpers/sca_code_scanner.py (hunk 31 above)
Comment (morsa4406) [reply cli/helpers/sca_code_scanner.py]: Fixed

File: cli/code_scanner.py (hunk 28 above)
Comment (morsa4406) [reply cli/code_scanner.py]: Fixed

File: cli/code_scanner.py [hunk 33]
```diff
@@ -341,6 +354,7 @@ def zip_documents_to_scan(scan_type: str, zip: InMemoryZip, documents: List[Docu
                      {'index': index, 'filename': document.path, 'unique_id': document.unique_id})
         zip.append(document.path, document.unique_id, document.content)
     zip.close()
+    logger.info("Zipped documents")
```
Comment (MichalBor) [inline cli/code_scanner.py:357]: delete

Comment (morsa4406) [reply cli/code_scanner.py]: Fixed

**Review by MichalBor — APPROVED** [review]
